8Knot/components/visualization.py

```
@@ -0,0 +1,93 @@
+                                    controls,
+                                    align="center",
+                                    justify="start",
+                                ),
+                            ]
+                        ),
+                    ],
+                    style={"padding": "1.5rem"},
+                ),
+            ],
+            className=class_name,
+            id=(nav_id if nav_id is not None else f"{self.page}-{self.viz_id}"),
```

Ruff (0.14.2)

Undefined name `nav_id`

(F821)